tests/components/knx/test_config_flow.py

```diff
@@ -999,6 +1005,100 @@ async def test_options_flow_connection_type(
         }
 
 
+async def test_options_flow_secure_manual_to_keyfile(hass: HomeAssistant) -> None:
+    """Test options flow changing secure credential source."""
+    mock_config_entry = MockConfigEntry(
+        title="KNX",
+        domain="knx",
+        data={
+            **DEFAULT_ENTRY_DATA,
+            CONF_KNX_CONNECTION_TYPE: CONF_KNX_TUNNELING_TCP_SECURE,
+            CONF_KNX_SECURE_USER_ID: 2,
+            CONF_KNX_SECURE_USER_PASSWORD: "password",
+            CONF_KNX_SECURE_DEVICE_AUTHENTICATION: "device_auth",
+            CONF_KNX_KNXKEY_FILENAME: "knx/testcase.knxkeys",
+            CONF_KNX_KNXKEY_PASSWORD: "invalid_password",
+            CONF_HOST: "192.168.0.1",
+            CONF_PORT: 3675,
+            CONF_KNX_INDIVIDUAL_ADDRESS: "0.0.240",
+            CONF_KNX_ROUTE_BACK: False,
+            CONF_KNX_LOCAL_IP: None,
+        },
+    )
+    mock_config_entry.add_to_hass(hass)
+    gateway = _gateway_descriptor(
+        "192.168.0.1",
+        3675,
+        supports_tunnelling_tcp=True,
+        requires_secure=True,
+    )
+
+    menu_step = await hass.config_entries.options.async_init(mock_config_entry.entry_id)
```

Comment:
We need to set up the config entry before starting the options flow to ensure that the config flow module is loaded.

Reply:
Isn't that what happens in 1028 ?
```py
mock_config_entry.add_to_hass(hass)
```

Reply:
No, that just adds it to the list of config entries.

Reply:
Ah, got it.
Quick question: Can these calls to `await hass.async_block_till_done()` after every step be removed (eg. L990)? Tests pass without them - so I'm not sure why they were added in the first place.
```py
    result = await hass.config_entries.flow.async_configure(
        result["flow_id"],
        {
            CONF_KNX_CONNECTION_TYPE: CONF_KNX_ROUTING,
        },
    )
    await hass.async_block_till_done()
```

Reply:
#82975

Reply:
If there may be a task created as a result of awaiting and continuing the flow we want to block till all tasks are done. If no tasks are created we don't need to block till done.

An example of a created task could be reloading the config entry when finishing the options flow, which happens in the background.